packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -14,8 +17,9 @@ import { InputValidator } from './validation';
 
 export class AssociationManager {
   public static associateProductWithPortfolio(
-    portfolio: IPortfolio, product: IProduct,
+    portfolio: IPortfolio, product: IProduct, options?: CommonConstraintOptions,
```

Comment:
To avoid the errors made during this refactoring, make this required, but possibly `undefined` instead:

```suggestion
    portfolio: IPortfolio, product: IProduct, options: CommonConstraintOptions | undefined,
```

This will prevent these errors.